// razorpayx-nodejs-sdk/dist/services/Contact/index.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
class RPXContact {
    constructor(client) {
        this.client = client;
    }
    /**
     * Creates a contact for the account
     * @link https://razorpay.com/docs/api/x/contacts#create-a-contact
     */
    create(contactInfo) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.client.load("/contacts", "POST", contactInfo);
        });
    }
    /**
     * updates a contact for the account
     * @link https://razorpay.com/docs/api/x/contacts#update-a-contact
     */
    update(contactId, contactInfo) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.client.load(`/contacts/${contactId}`, "PATCH", contactInfo);
        });
    }
    /**
     * Fetches all contacts the account
     * @link https://razorpay.com/docs/api/x/contacts#fetch-all-contacts
     */
    getAll(filters = {}) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.client.load("/contacts", "GET", filters);
        });
    }
    /**
     * Fetches details of a contact
     * @link https://razorpay.com/docs/api/x/contacts#fetch-a-contact-by-id
     */
    get(contactId) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.client.load(`/contacts/${contactId}`);
        });
    }
    /**
     * Activates a contact
     * @link https://razorpay.com/docs/api/x/contacts#activate-or-deactivate-a-contact
     */
    activate(contactId) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.client.load(`/contacts/${contactId}`, "PATCH", {
                active: true,
            });
        });
    }
    /**
     * Deactivates a contact
     * @link https://razorpay.com/docs/api/x/contacts#activate-or-deactivate-a-contact
     */
    deactivate(contactId) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.client.load(`/contacts/${contactId}`, "PATCH", {
                active: false,
            });
        });
    }
}
exports.default = RPXContact;
