// razorpayx-nodejs-sdk/dist/services/FundAccount/index.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
class RPXAccount {
    constructor(client) {
        this.client = client;
    }
    /**
     * Creates a fund account for a contact
     * @link https://razorpay.com/docs/api/x/fund-accounts#create-a-fund-account
     */
    create(accountInfo) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.client.load("/fund_accounts", "POST", accountInfo);
        });
    }
    /**
     * Fetches all fund accounts
     * @link https://razorpay.com/docs/api/x/fund-accounts#fetch-all-fund-accounts
     */
    getAll(filter = {}) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.client.load("/fund_accounts", "GET", filter);
        });
    }
    /**
     * Fetches details of an fund account
     * @link https://razorpay.com/docs/api/x/fund-accounts#fetch-fund-account-details-by-id
     */
    get(accountId) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.client.load(`/fund_accounts/${accountId}`);
        });
    }
    /**
     * validate the fund account
     * @link https://razorpay.com/docs/api/x/account-validation#validate-a-bank-account
     * @link https://razorpay.com/docs/api/x/account-validation#validate-a-vpa
     */
    validate(accountInfo) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.client.load("/fund_accounts/validations", "POST", accountInfo);
        });
    }
    /**
     * Activates a fund account
     * @link https://razorpay.com/docs/api/x/fund-accounts#activate-or-deactivate-a-fund-account
     */
    activate(accountId) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.client.load(`/fund_accounts/${accountId}`, "PATCH", {
                active: true,
            });
        });
    }
    /**
     * Deactivates a fund account
     * @link https://razorpay.com/docs/api/x/fund-accounts#activate-or-deactivate-a-fund-account
     */
    deactivate(accountId) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.client.load(`/fund_accounts/${accountId}`, "PATCH", {
                active: false,
            });
        });
    }
}
exports.default = RPXAccount;
