// razorpayx-nodejs-sdk/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RazorpayX = void 0;
const Contact_1 = __importDefault(require("./services/Contact"));
const FundAccount_1 = __importDefault(require("./services/FundAccount"));
const Payout_1 = __importDefault(require("./services/Payout"));
const PayoutLink_1 = __importDefault(require("./services/PayoutLink"));
const Transactions_1 = __importDefault(require("./services/Transactions"));
const RestClient_1 = __importDefault(require("./utils/RestClient"));
const RazorpayX = (key, secret, baseURL) => {
    if (!key || !secret) {
        throw new Error("Please pass API key and secret.");
    }
    const client = new RestClient_1.default({
        key,
        secret,
        baseURL,
    });
    return {
        Contact: new Contact_1.default(client),
        FundAccount: new FundAccount_1.default(client),
        Payout: new Payout_1.default(client),
        PayoutLink: new PayoutLink_1.default(client),
        Transactions: new Transactions_1.default(client),
    };
};
exports.RazorpayX = RazorpayX;
exports.default = exports.RazorpayX;
module.exports = exports.RazorpayX;
